adcc/ElectronicTransition.py

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env python3
+## vi: tabstop=4 shiftwidth=4 softtabstop=4 expandtab
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2020 by the adcc authors
+##
+## This file is part of adcc.
+##
+## adcc is free software: you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published
+## by the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## adcc is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with adcc. If not, see <http://www.gnu.org/licenses/>.
+##
+## ---------------------------------------------------------------------
+import warnings
+import numpy as np
+
+from .misc import cached_property
+from .timings import Timer, timed_member_call
+from .visualisation import ExcitationSpectrum
+from .OneParticleOperator import product_trace
+from .AdcMethod import AdcMethod
+
+from scipy import constants
+from matplotlib import pyplot as plt
+from .Excitation import mark_excitation_property
+from .solver.SolverStateBase import EigenSolverStateBase
+
+
+class ElectronicTransition:
+    def __init__(self, data, method=None, property_method=None):
+        """Construct an ElectronicTransition class from some data obtained
+        from an interative solver or other suitable data.
```

Comment:
"other suitable data" is a bit vague. I think I would put at least one more example.

Reply:
There are a few places where this occurs btw.

Reply:
Done.